### PR TITLE
Text editor

### PR DESCRIPTION
**Pull request by leshihak**
I really don't understand smth.
Can U help me with that?

### CODE REVIEW

**Review by vitaliyoliynykk — CHANGES_REQUESTED** [review]

File: src/components/NoteItem/NoteItem.tsx [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
+                        commodi rem officiis cum at voluptatem.
+                    </div>
                 </div>
-            );
-        });
+            </div>
+        );
     };
-    return <div className="container-note-item">{renderNotes()}</div>;
+    return <div className="container-note">{renderNotes()}</div>;
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:29]: This component should accept date title and description as props.Remove hardcoded values

File: src/components/NoteItemsList/NoteItemsList.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// .container-notes-list {
```
Comment (vitaliyoliynykk) [inline src/components/NoteItemsList/NoteItemsList.scss:1]: Remove comments

File: src/components/NoteItem/NoteItem.tsx [hunk 3]
```diff
@@ -1,28 +1,32 @@
 import React from 'react';
 import './NoteItem.scss';
-import { uuid } from 'uuidv4';
+import RemoveImg from '../../assets/remove.png';
+// import { uuid } from 'uuidv4';
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:4]: Remove comments

File: src/components/NoteItemsList/NoteItemsList.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
Comment (vitaliyoliynykk) [inline src/components/NoteItemsList/NoteItemsList.tsx:1]: What is this?

File: src/components/SearchInput/SearchInput.tsx [hunk 5]
```diff
@@ -1,12 +1,23 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './SearchInput.scss';
 import SearchIcon from '../../assets/search.png';
 
 const SearchInput = (): React.ReactElement => {
+    const [searchInputValue, setSearchInputValue] = useState('');
+    const getSearchInputValue = (event: React.FormEvent<HTMLInputElement>): void => {
+        setSearchInputValue(event.currentTarget.value);
```
Comment (vitaliyoliynykk) [inline src/components/SearchInput/SearchInput.tsx:8]: This component should call callback function to give value back to parent.

File: src/components/SearchInput/SearchInput.tsx [hunk 6]
```diff
@@ -1,12 +1,23 @@
...
+        console.log(searchInputValue);
```
Comment (vitaliyoliynykk) [inline src/components/SearchInput/SearchInput.tsx:9]: Remove console.log

File: src/components/TextEditor/TextEditor.tsx [hunk 7]
```diff
@@ -7,58 +7,47 @@ import AddFile from '../../assets/addfile.png';
...
         setStyles({ ...styles, fontSize: `${event.target.value}px` });
     };
 
-    const handleAlignTextLeft = (): void => {
-        setStyles({ ...styles, textAlign: 'left' });
-    };
-
-    const handleAlignTextCenter = (): void => {
-        setStyles({ ...styles, textAlign: 'center' });
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (vitaliyoliynykk) [inline src/components/TextEditor/TextEditor.tsx:32]: Do not use any

File: src/components/TextEditor/TextEditor.tsx [hunk 8]
```diff
@@ -7,58 +7,47 @@ import AddFile from '../../assets/addfile.png';
...
-            setTitleValue('');
-            setDescriptionValue('');
-        }
-    };
+    useEffect(() => {
+        setObjNote({
+            title: titleValue,
+            description: descriptionValue,
+            fontSize: `${fontSize}px`,
+            textAlign: '?',
```
Comment (vitaliyoliynykk) [inline src/components/TextEditor/TextEditor.tsx:42]: What does "?" mean?

File: src/components/TextEditor/TextEditor.tsx [hunk 9]
```diff
@@ -7,58 +7,47 @@ import AddFile from '../../assets/addfile.png';
...
+        });
+        console.log(objNote);
```
Comment (vitaliyoliynykk) [inline src/components/TextEditor/TextEditor.tsx:44]: Remove console.log

File: src/components/TextEditor/TextEditor.tsx [hunk 10]
```diff
@@ -7,58 +7,47 @@ import AddFile from '../../assets/addfile.png';
...
-interface Props {
-    getObjectNote: (obj: Note) => void;
-}
-interface Note {
-    title: string;
-    description: string;
-}
-
-const TextEditor = ({ getObjectNote }: Props): React.ReactElement => {
+const TextEditor = (): React.ReactElement => {
```
Comment (vitaliyoliynykk) [inline src/components/TextEditor/TextEditor.tsx:10]: Text editor should accept note object as a prop and populate it

**Review by vitaliyoliynykk — CHANGES_REQUESTED** [review]

File: src/components/TextEditor/TextEditor.tsx (hunk 7 above)
Comment (leshihak) [reply src/components/TextEditor/TextEditor.tsx:32]: Can you help me with the type of that function, pls

File: src/components/NoteItem/NoteItem.tsx [hunk 11]
```diff
@@ -1,32 +1,25 @@
...
 
-const NoteItem = (): React.ReactElement => {
-    const renderNotes = (): JSX.Element => {
+const NoteItem = ({ note }: { note: Note }): React.ReactElement => {
+    const renderNote = (): JSX.Element => {
         return (
-            <div className="container-note__block">
+            <>
                 <div className="container-note__header">
                     <div className="container-note__header_date">21 August 2020, 6:20 PM</div>
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:11]: should pass date as a prop

File: src/components/NoteItem/NoteItem.tsx [hunk 12]
```diff
@@ -1,32 +1,25 @@
...
                     <img src={RemoveImg} alt="#" className="container-note__header_img" />
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:12]: Clicking on remove should call callback function

File: src/components/NoteItem/NoteItem.tsx [hunk 13]
```diff
@@ -1,32 +1,25 @@
 import React from 'react';
 import './NoteItem.scss';
 import RemoveImg from '../../assets/remove.png';
-// import { uuid } from 'uuidv4';
+import { Note } from '../../models/models';
 
-const NoteItem = (): React.ReactElement => {
-    const renderNotes = (): JSX.Element => {
+const NoteItem = ({ note }: { note: Note }): React.ReactElement => {
+    const renderNote = (): JSX.Element => {
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:7]: function "renderNote" is not necessary

File: src/components/NoteItemsList/NoteItemsList.tsx [hunk 14]
```diff
@@ -1,31 +1,28 @@
 import React from 'react';
 import './NoteItemsList.scss';
+import NoteItem from '../NoteItem/NoteItem';
+import { v4 as uuidv4 } from 'uuid';
+import { Note } from '../../models/models';
 
 const NoteItemsList = (): React.ReactElement => {
```
Comment (vitaliyoliynykk) [inline src/components/NoteItemsList/NoteItemsList.tsx:7]: NoteItemsList should accept notes as a prop

File: src/components/TextEditor/TextEditor.tsx [hunk 15]
```diff
@@ -10,44 +10,39 @@ import AddList from '../../assets/list.png';
...
     };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const handleAlignText = (textAlign: any): void => {
-        setStyles({ ...styles, textAlign: textAlign });
+    const handleAlignText = (textAlign: 'left' | 'center' | 'right'): void => {
+        setStyles({ ...styles, textAlign });
     };
 
     useEffect(() => {
```
Comment (vitaliyoliynykk) [inline src/components/TextEditor/TextEditor.tsx:34]: You should call the callback function to give object up to parent

**Review by vitaliyoliynykk — CHANGES_REQUESTED** [review]

**Pull request merged**